pkgs/development/compilers/ponyc/default.nix

```
@@ -24,9 +24,9 @@ stdenv.mkDerivation (rec {
   };
 
   ponygbenchmark = fetchurl {
-    url = "https://github.com/google/benchmark/archive/v1.5.0.tar.gz";
-    sha256 = "06i2cr4rj126m1zfz0x1rbxv1mw1l7a11mzal5kqk56cdrdicsiw";
-    name = "v1.5.0.tar.gz";
+    url = "https://github.com/google/benchmark/archive/v1.5.2.tar.gz";
+    sha256 = "033hgsc76bcii1gwd8hdc7z1iqdnggk4f8cq0hzh98dsjsvxmjyw";
+    name = "v1.5.2.tar.gz";
```

that version should be parameterized and use fetchFromGitHub.

```suggestion
    url = "https://github.com/google/benchmark/archive/v1.5.2.tar.gz";
    sha256 = "033hgsc76bcii1gwd8hdc7z1iqdnggk4f8cq0hzh98dsjsvxmjyw";
    name = "ponygbenchmark-1.5.2.tar.gz";
```

---

Can you elaborate what parametrized means/how it looks like?

Also, I don't see how `fetchFromGitHub` improves things. Currently, `postUnpack` unpacks gbenchmark's source archive. `fetchFromGitHub` only provides a tree though, not an archive. The `tar` invocation could be changed to a `cp` of course. But then `postPatch` will fail, because it tries to patch shebangs, but the gbenchmark source has read-only permissions, as it's copied from nix store. That means we would need to fiddle around with permissions.

---

```suggestion
    url = "https://github.com/google/benchmark/archive/v{version}.tar.gz";
    sha256 = "033hgsc76bcii1gwd8hdc7z1iqdnggk4f8cq0hzh98dsjsvxmjyw";
    name = "v{version}.tar.gz";
```